### DISCUSSION TITLE
theia build --mode development , An error occurred , why ?

### DISCUSSION THREAD

**Discussion by 94Rabbit** [eclipse-theia/theia#10534] [Q&A]
![image](https://user-images.githubusercontent.com/25635994/145722259-5483b704-68a3-41b8-945d-855f9ad3c5e7.png)

**Comment by msujew** [comment]
Hi @LhLoveZgg,

do you mind giving us a bit more context around your setup? Just the error message isn't enough to tell what's wrong.

For example, did you modify your `webpack.config.js` compared to the default setup (which is just using the generated webpack config) in any way? Additionally are you using the latest version of Theia? (1.20.0)

**Reply by 94Rabbit** [reply]
{
  "name": "hopper",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "theia build --mode development",
    "download": "theia download:plugins",
    "clean": "theia clean && rm -rf node_modules && rm -rf webpack.config.js && rm -rf plugins",
    "start": "theia start"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "theia": {
    "next": "true",
    "target": "electron",
    "frontend": {
      "config": {
        "applicationName": "hopper",
        "preferences": {
          "files.enableTrash": false
        },
        "defaultTheme": "light",
        "defaultIconTheme": "vs-seti",
        "electron": {
          "disallowReloadKeybinding": true,
          "windowOptions": {
            "titleBarStyle": "hidden",
            "webPreferences": {
              "webSecurity": false,
              "nodeIntegration": true,
              "webviewTag": true
            }
          }
        }
      }
    }
  },
  "dependencies": {
    "@theia/electron": "^1.20.0"
  },
  "theiaPluginsDir": "plugins",
  "theiaPlugins": {
    "vscode-editorconfig": "https://open-vsx.org/api/EditorConfig/EditorConfig/0.14.4/file/EditorConfig.EditorConfig-0.14.4.vsix"
  },
  "theiaPluginsExcludeIds": [
    "vscode.cpp"
  ],
  "devDependencies": {
    "@theia/cli": "^1.21.0-next.18"
  }
}

**Reply by 94Rabbit** [reply]
> Hi @LhLoveZgg,
> 
> do you mind giving us a bit more context around your setup? Just the error message isn't enough to tell what's wrong.
> 
> For example, did you modify your `webpack.config.js` compared to the default setup (which is just using the generated webpack config) in any way? Additionally are you using the latest version of Theia? (1.20.0)

my package.json
{
"name": "hopper",
"version": "1.0.0",
"description": "",
"main": "index.js",
"scripts": {
"test": "echo "Error: no test specified" && exit 1",
"build": "theia build --mode development",
"download": "theia download:plugins",
"clean": "theia clean && rm -rf node_modules && rm -rf webpack.config.js && rm -rf plugins",
"start": "theia start"
},
"keywords": [],
"author": "",
"license": "ISC",
"theia": {
"next": "true",
"target": "electron",
"frontend": {
"config": {
"applicationName": "hopper",
"preferences": {
"files.enableTrash": false
},
"defaultTheme": "light",
"defaultIconTheme": "vs-seti",
"electron": {
"disallowReloadKeybinding": true,
"windowOptions": {
"titleBarStyle": "hidden",
"webPreferences": {
"webSecurity": false,
"nodeIntegration": true,
"webviewTag": true
}
}
}
}
}
},
"dependencies": {
"@theia/electron": "^1.20.0"
},
"theiaPluginsDir": "plugins",
"theiaPlugins": {
"vscode-editorconfig": "https://open-vsx.org/api/EditorConfig/EditorConfig/0.14.4/file/EditorConfig.EditorConfig-0.14.4.vsix"
},
"theiaPluginsExcludeIds": [
"vscode.cpp"
],
"devDependencies": {
"@theia/cli": "^1.21.0-next.18"
}
}

**Comment by 94Rabbit** [comment]
![image](https://user-images.githubusercontent.com/25635994/145723290-2af8ffa6-8e17-4562-9d6d-d41e57ef191b.png)

**Reply by msujew** [reply]
I see, that's interesting. Usually, Theia would add the monaco components in there, like this:
```js
patterns: [{
  from: monacoEditorCorePath,
  to: 'vs'
}]
```

Do you have the `@theia/monaco` component in your browser/electron-apps `package.json` as a dependency? Do you mind adding it to see whether it fixes the behavior?

**Comment by msujew** [comment]
@LhLoveZgg I identified the issue, documented it in #10535 and provided a fix #10536. As a workaround, you can add `@theia/monaco` as a dependency to your app.

**Reply by 94Rabbit** [reply]
i will try later, thanks

**Reply by 94Rabbit** [reply]
@theia/monaco is necessary?

**Reply by msujew** [reply]
At least right now, yes. The PR will address this, so it won't be necessary in the future.
